lib/src/store.dart

```diff
@@ -83,25 +89,46 @@ class Store {
   /// If [onAction] is provided, it will be called every time [action] is
   /// dispatched. If [onAction] returns a [Future], [trigger] will not be
   /// called until that future has resolved.
+  ///
+  /// If the `Store` has been disposed, this method throws a [StateError].
   triggerOnAction(Action action, [void onAction(payload)]) {
+    if (isDisposed) {
+      throw new StateError('Store has been disposed');
+    }
+
     if (onAction != null) {
-      action.listen((payload) async {
+      var sub = action.listen((payload) async {
         await onAction(payload);
         trigger();
       });
+      manageDisposer(() => new Future(() {
+            sub.cancel();
+          }));
```

Comment:
I never use `triggerOnAction`. Can we have a utility method like this https://github.com/Workiva/w_common/blob/master/lib/disposable.dart#L169? Something like `manageActionSubscription`?